share/Modules/Preproc.cmake

```diff
@@ -0,0 +1,113 @@
+#
+#  Preproc.cmake (runtime)
+#  Copyright 2020 ItJustWorksTM
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+function (identify_source SOURCE_FILE SOURCE_TYPE)
+    set (SOURCE_FILE_ORIGINAL "${SOURCE_FILE}" PARENT_SCOPE)
+    if (SOURCE_TYPE STREQUAL "sketch")
+        if (SOURCE_FILE MATCHES "\\.ino$" OR SOURCE_FILE MATCHES "\\.pde$" )
+            set (SOURCE_IS_CPP OFF PARENT_SCOPE)
+            set (SOURCE_IS_CPP OFF)
+            set (SOURCE_IS_INO ON PARENT_SCOPE)
+            set (SOURCE_IS_INO ON)
```

Comment:
It looks like the code is repeated in lines 32-35 and 39-42. If there isn't a way to change it, not a problem. Everything else looks pretty good.

Reply:
The only ways to deduplicate these wouldn't make the code any more sane to read (quite the opposite I'm afraid)